crates/rust-client/src/store/note_record/input_note_record.rs

```
@@ -32,277 +32,306 @@ use crate::ClientError;
+    /// status
+    pub fn is_authenticated(&self) -> bool {
+        matches!(
+            self.state,
+            NoteState::Committed { .. }
+                | NoteState::ProcessingAuthenticated { .. }
+                | NoteState::NativeConsumed { .. }
+        )
     }
 
-    pub fn ignored(&self) -> bool {
-        self.ignored
+    // TRANSITIONS
+    // ================================================================================================
+
+    pub fn inclusion_proof_received(
+        &mut self,
+        inclusion_proof: NoteInclusionProof,
+        metadata: NoteMetadata,
+    ) -> Result<bool, NoteRecordError> {
+        match &self.state {
+            // Note had no inclusion proof
+            NoteState::Expected { .. } | NoteState::Unknown => {
+                self.state = NoteState::Unverified { inclusion_proof, metadata };
+                Ok(true)
+            },
+            // Note had an inclusion proof
+            NoteState::Unverified {
+                metadata: old_metadata,
+                inclusion_proof: old_inclusion_proof,
+            }
+            | NoteState::Committed {
+                metadata: old_metadata,
+                inclusion_proof: old_inclusion_proof,
+                ..
+            }
+            | NoteState::ProcessingAuthenticated {
+                metadata: old_metadata,
+                inclusion_proof: old_inclusion_proof,
+                ..
+            } => {
+                if old_inclusion_proof != &inclusion_proof || old_metadata != &metadata {
+                    return Err(NoteRecordError::StateTransitionError(
+                        "Inclusion proof or metadata do not match the expected values".to_string(),
+                    ));
+                }
+                Ok(false)
+            },
+            _ => todo!("How should we deal with invalid/unnecessary transitions? Maybe we don't want a default '_' case"),
```

If a transition is invalid we should return an error. But if a transition has no effect - e.g., we got an inclusion proof for a consumed note, we should just return `Ok(false)`, I think.